File: compiler/testData/loadJava/compiledJava/kotlinSignature/propagation/parameter/Kt3302.java
package test;

import org.jetbrains.annotations.NotNull;

// See SubclassFromGenericAndNot, as well
public interface Kt3302 {
    public interface BSONObject {
        Object put(@NotNull String s, @NotNull Object o);

        void dummy(); // to avoid loading as SAM interface
    }

    public interface LinkedHashMap<K, V> {
        public V put(K key, V value);

        void dummy(); // to avoid loading as SAM interface
    }

    public interface BasicBSONObject extends LinkedHashMap<String, Object>, BSONObject {
        @Override
        public Object put(String key, Object value);
    }
}

File: compiler/testData/loadJava/compiledKotlin/fromLoadJava/kotlinSignature/propagation/parameter/Kt3302.kt
package test

// See SubclassFromGenericAndNot, as well
public interface Kt3302 {
    public interface BSONObject  {
        public fun put(p0: String, p1: Any): Any?

        public fun dummy() // to avoid loading as SAM interface
    }

    public interface LinkedHashMap<K, V>  {
        public fun put(key: K, value: V): V?

        public fun dummy() // to avoid loading as SAM interface
    }

    public interface BasicBSONObject : LinkedHashMap<String, Any>, BSONObject {
        override fun put(key: String, value: Any): Any?
    }
}
